qadence/extensions.py

```diff
@@ -60,6 +63,18 @@ def _validate_diff_mode(backend: Backend, diff_mode: DiffMode) -> None:
     """Fallback function for native Qadence diff_mode if extensions is not present."""
     if not backend.supports_ad and diff_mode == DiffMode.AD:
         raise TypeError(f"Backend {backend.name} does not support diff_mode {DiffMode.AD}.")
+    elif not backend.supports_adjoint and diff_mode == DiffMode.ADJOINT:
+        raise TypeError(f"Backend {backend.name} does not support diff_mode {DiffMode.ADJOINT}.")
+
+
+def _validate_backend_config(backend: Backend) -> None:
+    if backend.config.use_gradient_checkpointing:
+        msg = "use_gradient_checkpointing will soon be deprecated.\
+                                        To save memory, try diff_mode=DiffMode.ADJOINT instead."
```

Comment:
Well, it is deprecated already, right? I mean that if you choose it, it won't do anything. I would say that in the message.

Reply:
it is now